src/compiler/parser.ts

```diff
@@ -2399,7 +2405,7 @@ namespace Parser {
         }
 
         // The user alternatively might have misspelled or forgotten to add a space after a common keyword.
-        const suggestion = getSpellingSuggestion(expressionText, viableKeywordSuggestions, n => n) ?? getSpaceSuggestion(expressionText);
+        const suggestion = getSpellingSuggestion(expressionText, viableKeywordSuggestions, identity) ?? getSpaceSuggestion(expressionText);

```

Comment:
This change seems to be unrelated to the rest of the PR.

Reply:
This is because I did the same in `scanCharacterClassEscape`